src/main/java/cz/it4i/parallel/ui/LastFormLoader.java

```diff
@@ -0,0 +1,76 @@
+package cz.it4i.parallel.ui;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Base64;
+
+import org.scijava.plugin.Parameter;
+import org.scijava.prefs.PrefService;
+
+import cz.it4i.parallel.runners.ImageJServerRunnerSettings;
+import lombok.extern.slf4j.Slf4j;
+
+//Code responsible for loading and storing the last used form values:
+@Slf4j
+public class LastFormLoader {
+	
+	private String formStorage;
+	
+	@Parameter
+	private PrefService prefService;
+	
+	public LastFormLoader(PrefService newPrefService, String newFormName) {
+		this.prefService = newPrefService;
+		this.formStorage = newFormName;
+	}
+
+	public void storeLastForm(Object form) {
+		String serializedForm = serializeForm(form);
+		prefService.put(this.getClass(), formStorage, serializedForm);
+	}
+
+	public Object loadLastForm() {
```

Comment:
It could return directly subtype of RunnerSettings - LastFormLoader will be generic type.

Reply:
Changed Object to RunnerSettings in LastFormLoader.

Reply:
I meant that LastFormLoader will be generic type - look at the commit https://github.com/fiji-hpc/scijava-parallel/commit/db06de467fdc99cdbd869b22490fada262dc0eca .